src/common/services/__specs__/git.spec.js

```diff
@@ -208,73 +216,65 @@ describe('services/git', () => {
           postCommitExists = true
         })
 
-        it('merges git-switch call into post-commit', () => {
+        it('merges git-switch call into post-commit', async () => {
           existingPostCommitScript = '#!/bin/bash\n\necho "Committed"'
           const expected = `${subject.POST_COMMIT_BASE}\n\necho "Committed"`
 
-          subject.initRepo(repoPath)
+          await subject.initRepo(repoPath)
 
           expect(fs.writeFileSync).to.have.been.calledWith(postCommitPath, expected, { encoding: 'utf-8', mode: 0o755 })
         })
       })
 
       describe('when sub-modules exist', () => {
         let submoduleDirs
+        let submoduleStatus
         let submodule1GitHooksPath
         let submodule2GitHooksPath
+        let submodule3GitHooksPath
 
         beforeEach(() => {
           submoduleExists = true
-          submoduleDirs = ['mod1', 'mod2']
+          submoduleDirs = ['mod1', 'subdir/mod2', 'subdir/mod3']
+          submoduleStatus = submoduleDirs
+            .map((dir, i) => `${i % 2 === 0 ? '+' : ' '}rando-commit-hash ${dir} (current/branch)`)
```

Comment:
Not quite sure I fully understand the `i % 2` step.

Reply:
That adds a `+` sign to the front of even submodules. That is so we can test that the parsing works for both submodules that have been initialized (preceding `+`) and those that haven't.